nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/blob/ColdStorageHelper.java

```diff
@@ -148,7 +154,17 @@ public static ColdStorageContentStatus checkAvailabilityOfColdStorageContent(Cor
         for (DocumentModel doc : documents) {
             if (isColdStorageContentAvailable(doc)) {
                 available++;
+                // mark the cold storage content of document as retrieved
+                doc.setPropertyValue(COLD_STORAGE_BEING_RETRIEVED_PROPERTY, false);
+                session.saveDocument(doc);
```

Comment:
We might need a `session.save()` here, though the transactional context might be enough (we're in a synchronous listener called by a scheduled event), not sure.
@efge WDYT?